### PR TITLE
SAK-44880: SAMIGO - Error on log when printing PDF assessment that contains a BASE64 image

### PR DESCRIPTION
**Pull request by jesusmmp**

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java [hunk 1]
```diff
@@ -176,6 +179,23 @@ else if (src.startsWith("temp://")) {
 					log.error(e.getMessage(), e);
 				}
 			}
+			else if (src.startsWith("data:image/")) {
+				Image img = null;
+				final String base64Data = src.substring(src.indexOf(",") + 1);
+				try {
+					img = Image.getInstance(Base64.decode(base64Data));
+				} catch (Exception e) {
+					log.error(e.getMessage(), e);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java:188]: The log statement here should be more descriptive, and should it be a warning vs an error?

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java [hunk 2]
```diff
@@ -176,6 +179,23 @@ else if (src.startsWith("temp://")) {
...
+				}
+				if (img != null) {
+					try {
+						document.add(img);
+					}
+					catch (Exception e) {
+						log.error(e.getMessage(), e);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java:195]: same uninformative message

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java (hunk 1 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java]: Just changed!

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java (hunk 2 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java]: Just changed!

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java [hunk 3]
```diff
@@ -176,6 +179,23 @@ else if (src.startsWith("temp://")) {
 					log.error(e.getMessage(), e);
 				}
 			}
+			else if (src.startsWith("data:image/")) {
+				Image img = null;
+				final String base64Data = src.substring(src.indexOf(",") + 1);
+				try {
+					img = Image.getInstance(Base64.decode(base64Data));
+				} catch (Exception e) {
+					log.warn("Failed retrieving image", e);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java:188]:
```suggestion
					log.warn("Failed retrieving image", e.toString());
```

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java [hunk 4]
```diff
@@ -176,6 +179,23 @@ else if (src.startsWith("temp://")) {
...
+				}
+				if (img != null) {
+					try {
+						document.add(img);
+					}
+					catch (Exception e) {
+						log.warn("Image couldn't be added to the document", e);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/HTMLWorker.java:195]:
```suggestion
						log.warn("Image couldn't be added to the document", e.toString());
```

**Review by ern — APPROVED** [review]

**Comment by jesusmmp** [comment]
Thanks @ern ! :)

**Pull request merged**